kornia/feature/mkd.py

```diff
@@ -310,7 +314,7 @@ def __init__(self,
 
         # Store precomputed embedding.
         self.register_buffer('emb', emb.unsqueeze(0))
-        self.d_emb: int = self.emb.shape[1]  # type: ignore
+        self.d_emb: int = emb.shape[0]
```

Comment:
Are you sure that intend to change index?

Reply:
there's an issue with registered buffers and typing.

Typing in pt == 1.6 doesn't seem to recognize the retrieved buffer in it's original type and assumes that could be `torch.Tensor`, `nn.Module` or `Any`.

This is why mypy doesn't recognize the variable to have a shape member. `self.emb` has shape `(1,D,...)` because of the unsqueeze from `emb`. I'm assuming here `emb.shape[0] == self.emb.shape[1]` after registering the buffer.